workspaces/ui-v2/src/pages/docs/components/DataFetchers.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { FC } from 'react';
+import { createFlatList } from '<src>/components/FieldOrParameter';
+import { useShapeDescriptor } from '<src>/hooks/useShapeDescriptor';
+import { IShapeRenderer } from '<src>/components';
+
+type ContributionFetcherProps = {
+  rootShapeId: string;
+  endpointId: string;
+  // TODO QPB change this typing - currently this holds contributions, and the data for rendering the shape
```

Comment:
This is confusing because we have two typings for contributions - `IContribution` - which is raw contribution information and `IContributions` which is a mash of contribution, and field data. The `FieldOrParameter` component needs to be changed (i.e. a more scoped interface) in order for us to remove this adhoc typing that is difficult to reason about